### PR TITLE
Meleespeech Action, Holoparasite Meleespeech

### PR DESCRIPTION
**Pull request by HerCoyote23**
Previous PR, #18501 got stuck in git bash mistake hell. 
## About the PR

Makes the MeleeSpeech component have an action to configure your battlecry.
Adds MeleeSpeech to Holoparasites.

Be advised that this is my first foray into actions and I was very much figuring out how they work over this PR. It's likely a bit flawed in implementation.
## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: The Syndicate's Holoparasite have taken to yelling catchphrases as they fight.

### CODE REVIEW

File: Content.Shared/Speech/Components/MeleeSpeechComponent.cs [hunk 1]
```diff
@@ -1,21 +1,40 @@
-using Content.Shared.Speech.EntitySystems;
+using Content.Shared.Actions;
+using Content.Shared.Actions.ActionTypes;
+using Robust.Shared.GameStates;
 using Robust.Shared.Serialization;
-
 namespace Content.Shared.Speech.Components;
-
-[RegisterComponent]
+[RegisterComponent, NetworkedComponent]
```
Comment (metalgearsloth) [inline Content.Shared/Speech/Components/MeleeSpeechComponent.cs:6]: newline

File: Content.Shared/Speech/Components/MeleeSpeechComponent.cs [hunk 2]
```diff
@@ -1,21 +1,40 @@
...
-	public string? Battlecry;
-
+    /// <summary>
+    /// The battlecry to be said when an entity attacks with this component
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField("Battlecry")]
+    [AutoNetworkedField]
+    public string? Battlecry;
+    /// <summary>
```
Comment (metalgearsloth) [inline Content.Shared/Speech/Components/MeleeSpeechComponent.cs:18]: newline

File: Content.Shared/Speech/Components/MeleeSpeechComponent.cs [hunk 3]
```diff
@@ -1,21 +1,40 @@
...
+    /// The maximum amount of characters allowed in a battlecry
+    /// </summary>
     [ViewVariables(VVAccess.ReadWrite)]
     [DataField("MaxBattlecryLength")]
+    [AutoNetworkedField]
     public int MaxBattlecryLength = 12;
+    /// <summary>
```
Comment (metalgearsloth) [inline Content.Shared/Speech/Components/MeleeSpeechComponent.cs:25]: newline

File: Content.Server/Speech/EntitySystems/MeleeSpeechSystem.cs [hunk 4]
```diff
@@ -1,76 +1,83 @@
...
+            battlecry = battlecry.Trim();
+        }
+        else
+        {
+            battlecry = null;
+        }
+        if (meleeSpeechComp.Battlecry == battlecry)
+            return true;
+        meleeSpeechComp.Battlecry = battlecry;
+        Dirty(meleeSpeechComp);
```
Comment (metalgearsloth) [inline Content.Server/Speech/EntitySystems/MeleeSpeechSystem.cs:79]:
```suggestion
        Dirty(uid, meleeSpeechComp);
```

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
99% there I think just some minor stuff gj.

**Comment by HerCoyote23** [comment]
Think Holoclowns should have meleespeech as well? Atm they inherit it from Guardian, don't know if it's quite thematically fitting.

**Comment by VektorZ1** [comment]
The holoclowns should make a honk sound when struck :trollface:

**Review by keronshb — APPROVED** [review]

**Pull request merged**